include/makefile.in

```diff
@@ -297,18 +297,18 @@ dist-devel-post: dist-devel-recursive
 dist-devel-recursive: dist-devel-pre
 
 stamp:
-	@cp stamp.h stamp2.h; \
-	stamp_version=\"$(PACKAGE_VERSION)\"; \
-	if [ -e $(rootfromhere)/.git ]; then \
-	  stamp_version=\"`$(GIT) describe --tag --always 2> /dev/null | sed -e s/-bootstrap$$//`\"; \
-	fi; \
-	if fgrep "#define ___STAMP_VERSION $$stamp_version" stamp.h > /dev/null; then \
-	  echo "*** include/stamp.h untouched (___STAMP_VERSION did not change)"; \
-	else \
-	  echo "*** include/stamp.h updated to have ___STAMP_VERSION=$$stamp_version"; \
-	  eval `date -u '+stamp_ymd=%Y%m%d stamp_hms=%H%M%S'`; \
-	  sed -e "s/#define ___STAMP_YMD [0-9][0-9]*/#define ___STAMP_YMD `expr 0 + $$stamp_ymd`/" -e "s/#define ___STAMP_HMS [0-9][0-9]*/#define ___STAMP_HMS `expr 0 + $$stamp_hms`/" -e "s/#define ___STAMP_VERSION .*/#define ___STAMP_VERSION $$stamp_version/" $(srcdirpfx)stamp.h > stamp2.h; \
-	  mv stamp2.h stamp.h; \
+	echo > stamp.h && \
+	if [ -e "$(rootfromhere)/.git" ]; then \
+	    version="$$($(GIT) describe --tag --always | sed 's/-bootstrap$$//')" && \
+	    ymd="$$(expr 0 + "$$(date -u '+%Y%m%d')")" && \
+	    hms="$$(expr 0 + "$$(date -u '+%H%M%S')")" && \
+	    { \
+	        echo "/* Generated from scratch for each build */" && \
+	        echo "#define ___STAMP_VERSION \"$$version\"" && \
```

Comment:
For consistency with `stamp-release.h` and allow forcing a specific stamp through `gcc -D___STAMP_VERSION=...` the generated code should be wrapped in `#ifndef`s:

```
echo "#ifndef ___STAMP_VERSION" && \
echo "#define ___STAMP_VERSION \"$$version\"" && \
echo "#endif" && \
echo "" && \
```

and similarly for the others.

Reply:
OK, done.